packages/openapi-fetch/src/index.d.ts

```diff
@@ -39,6 +39,15 @@ export type QuerySerializer<T> = (
 export type BodySerializer<T> = (body: OperationRequestBodyContent<T>) => any;
 
 export type ParseAs = "json" | "text" | "blob" | "arrayBuffer" | "stream";
+export type ParseAsResponse<T, K> = K extends ParseAs
+  ? {
+      json: T;
+      text: Awaited<ReturnType<Response["text"]>>;
+      blob: Awaited<ReturnType<Response["blob"]>>;
+      arrayBuffer: Awaited<ReturnType<Response["arrayBuffer"]>>;
+      stream: Response["body"];
+    }[K]
```

Comment:
👍 This is a smart pattern. We could probably use this in other places to simplify some deeply-nested ternaries (maybe).